File: Exercism Codes/Python/binary_search.py
def find(search_list, value):
    left = 0
    right = len(search_list) - 1
    while left <= right:
        mid = left + ((right - left) // 2)

        if search_list[mid] == value:
            return mid

        if search_list[mid] < value:
            left = mid + 1

        if search_list[mid] > value:
            right = mid - 1

    raise ValueError("value not in array")

File: Exercism Codes/JavaScript/binary-search.js
export const find = (arr, el) => {
  let left = 0;
  let right = arr.length - 1;

  while (left <= right) {
    const mid = left + Math.floor((right - left) / 2);

    if (arr[mid] === el) {
      return mid;
    }

    if (arr[mid] > el) {
      right = mid - 1;
    }

    if (arr[mid] < el) {
      left = mid + 1;
    }
  }

  throw new Error('Value not in array');
};
